processor/src/main/java/org/mapstruct/ap/internal/model/source/BeanMapping.java

```diff
@@ -61,12 +63,25 @@ public static BeanMapping fromPrism(BeanMappingPrism beanMapping, ExecutableElem
             beanMapping.qualifiedByName(),
             resultTypeIsDefined ? beanMapping.resultType() : null );
 
-        return new BeanMapping(cmp, nullValueMappingStrategy );
+        //TODO Do we want to add the reporting policy to the BeanMapping as well? To give more granular support?
```

Comment:
If I understand you correctly, you now just delegate it down to the @MappingConfig, to the @Mapper to the @BeanMapping? 

I (vaguely) remember that I discussed this earlier somehow with @gunnarmorling ... He was not in favor at that point in time. However, it could be another strategy...